src/stcal/resample/resample.py

```diff
@@ -1014,80 +1017,42 @@ def resample_variance_arrays(self, model, pixmap, iscale,
-            if self.weight_type.startswith("ivm") and rn_var is not None:
-                weight = np.ones(self.output_array_shape)
-                weight[mask] = np.reciprocal(rn_var[mask])
-
-            elif self.weight_type == "exptime":
-                t, _ = get_tmeasure(model)
-                weight = np.full(self.output_array_shape, t)
-
-            # Weight and add the readnoise variance
-            # Note: floating point overflow is an issue if variance weights
-            # are used - it can't be squared before multiplication
-            if rn_var is not None:
-                # Add the inverse of the resampled variance to a running sum.
-                # Update only pixels (in the running sum) with
-                # valid new values:
-                mask = (rn_var >= 0) & np.isfinite(rn_var) & (weight > 0)
-                self._var_rnoise_wsum[mask] = np.nansum(
-                    [
-                        self._var_rnoise_wsum[mask],
-                        rn_var[mask] * weight[mask] * weight[mask]
-                    ],
-                    axis=0
-                )
-                self._var_rnoise_weight[mask] += weight[mask]
-
-        # Now do poisson and flat variance, updating only valid new values
-        # (zero is a valid value; negative, inf, or NaN are not)
-        if self._check_var_array(model, "var_poisson"):
-            pn_var = self._resample_one_variance_array(
-                "var_poisson",
-                model=model,
-                **pars,
-            )
-            if pn_var is not None:
-                mask = (pn_var >= 0) & np.isfinite(pn_var) & (weight > 0)
-                self._var_poisson_wsum[mask] = np.nansum(
-                    [
-                        self._var_poisson_wsum[mask],
-                        pn_var[mask] * weight[mask] * weight[mask]
-                    ],
-                    axis=0
+        for varname in self._enable_var:
+            if self._check_var_array(model, varname):
+                var = self._resample_one_variance_array(
+                    varname,
+                    model=model,
+                    **pars,
                 )
-                self._var_poisson_weight[mask] += weight[mask]
+                self.variance_info[varname]['var'] = var
 
-        if self._check_var_array(model, "var_flat"):
-            flat_var = self._resample_one_variance_array(
-                "var_flat",
-                model=model,
-                **pars,
-            )
-            if flat_var is not None:
-                mask = (flat_var >= 0) & np.isfinite(flat_var) & (weight > 0)
-                self._var_flat_wsum[mask] = np.nansum(
+        # Set the weight for the image from the weight type
+        if self.weight_type.startswith("ivm"):
+            rn_var = self.variance_info['var_rnoise']['var']
+            mask = (rn_var > 0) & np.isfinite(rn_var)
+            weight = np.zeros(self.output_array_shape, dtype='f4')
```

Comment:
This was `float64` in the original code, I believe. I think for these computations we want increased precision for intermediate computations.